scripts/builtin/csplineCG.dml

```diff
@@ -0,0 +1,275 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+#
+# THIS SCRIPT SOLVES CUBIC SPLINE INTERPOLATION USING THE CONJUGATE GRADIENT ALGORITHM
+#
+# INPUT PARAMETERS:
+# --------------------------------------------------------------------------------------------
+# NAME  TYPE           DEFAULT   MEANING
+# --------------------------------------------------------------------------------------------
+# X     Matrix[Double]  ---      1-column matrix of x values knots
+# Y     Matrix[Double]  ---      1-column matrix of corresponding y values knots
+# inp_x Double          ---      the given input x, for which the cspline will find predicted y.
+# Log   String          " "      Location to store iteration-specific variables for monitoring and debugging purposes
+#
+# tol   Double          0.000001 Tolerance (epsilon); conjugate graduent procedure terminates early if
+#                                L2 norm of the beta-residual is less than tolerance * its initial norm
+# maxi  Int             0        Maximum number of conjugate gradient iterations, 0 = no maximum
+# --------------------------------------------------------------------------------------------
+# OUTPUT: 
+# pred_Y Matrix[Double] ---      Predicted value
+# K      Matrix[Double] ---      Matrix of k parameters
+#
+# The Log file, when requested, contains the following per-iteration variables in CSV
+# format, each line containing triple (NAME, ITERATION, VALUE) with ITERATION = 0 for
+# initial values:
+#
+# NAME                  MEANING
+# -------------------------------------------------------------------------------------
+# CG_RESIDUAL_NORM      L2-norm of Conj.Grad.residual, which is A %*% beta - t(X) %*% y
+#                           where A = t(X) %*% X + diag (lambda), or a similar quantity
+# CG_RESIDUAL_RATIO     Ratio of current L2-norm of Conj.Grad.residual over the initial
+# -------------------------------------------------------------------------------------
+#
+
+#Assumptions:
+# - The inputs xs are monotonically increasing,
+# - there is no duplicates points in x
+
+#Algorithms: It implement the https://en.wikipedia.org/wiki/Spline_interpolation#Algorithm_to_find_the_interpolating_cubic_spline
+#it usages natural spline with q1''(x0) == qn''(xn) == 0.0
+
+
+m_csplineCG =function (Matrix[Double] X, Matrix[Double] Y, Double inp_x, 
+  Double tol = 0.000001, Integer maxi = 10
+  ) 
+  return (Matrix[Double] pred_Y, Matrix[Double] K) 
+{
+  xs = X;
+  ys = Y;
+  inp_x = inp_x;
+  fileLog = " ";     
```

Comment:
There is no way of setting this file Log in parameters.
I would suggest either removing the feature, or replacing it with print statements. (surrounded by verbose flags.)